### PR TITLE
[WIP][#3310] Display Spatial Coverage Place/Area Name in Edit mode.

### PR DESCRIPTION
**Pull request by Maurier**
This was caused by an asynchronous call that updates the coverage when a folder structure is loaded in the file browser of a resource. Fixed by adding the name value to the dictionary returned in AJAX request. 

### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Create a resource.
2. Add spatial coverage. Make sure you fill in the value for "Place/Area name".
3. Reload the page and verify "Place/Area name" displays correctly in both View and Edit mode.

### CODE REVIEW

**Comment by sblack-usu** [comment]
retest this, please

**Comment by Maurier** [comment]
I will be updating a failing test before merging this.

File: hs_core/views/utils.py [hunk 1]
```diff
@@ -1290,6 +1290,7 @@ def get_coverage_data_dict(source, coverage_type='spatial'):
         spatial_coverage_dict = {}
         if spatial_coverage:
             spatial_coverage_dict['type'] = spatial_coverage.type
+            spatial_coverage_dict['name'] = spatial_coverage.name
```
Comment (pkdash) [inline hs_core/views/utils.py]: Should be: `spatial_coverage_dict['name'] = spatial_coverage.value['name']`

File: hs_core/views/utils.py [hunk 2]
```diff
@@ -1290,6 +1290,7 @@ def get_coverage_data_dict(source, coverage_type='spatial'):
         spatial_coverage_dict = {}
         if spatial_coverage:
             spatial_coverage_dict['type'] = spatial_coverage.type
+            spatial_coverage_dict['name'] = spatial_coverage.value['name']
```
Comment (pkdash) [inline hs_core/views/utils.py]: It is possible that 'name' may be missing in coverage data. So it needs to be:
`spatial_coverage_dict['name'] = spatial_coverage.value.get('name', "") `

Comment (Maurier) [reply hs_core/views/utils.py]: This fixed the test case. Thanks!

**Comment by Maurier** [comment]
@engineSound, the failing test case has been updated. Please review when you get a chance.

File: theme/templates/resource-landing-page/coverage.html [hunk 3]
```diff
@@ -24,6 +24,8 @@ <h4>Spatial</h4>
                         {% if spatial_coverage.name %}
                             <div>
                                 <small>Place/Area Name: </small>
+                            </div>
+                            <div class="space-bottom">
```
Comment (engineSound) [inline theme/templates/resource-landing-page/coverage.html]: Well done. I noticed the space-bottom in the hydroshare-core.css.
.space-bottom {
	margin-bottom: 15px;
}

**Pull request merged**